library/classes/thumbnail/Thumbnail.class.php

```diff
@@ -0,0 +1,178 @@
+<?php
+/**
+ * Make thumbnail file by PHP GD functions.
+ * version 1 - just for images.
+ *
+ * Another: Amiel Elboim
+ * Date: 29/05/16
+ */
+class Thumbnail
+{
+
+    //max size of thumbnail
+    public $max_size = 100;
+    public $thumbnail_type = 'jpg';
+    /**
+     * Enable to set max size of thumbnail
+     * @param (int) $max_size
+     */
+    public function __construct($max_size = null)
+    {
+        if(!is_null($max_size))$this->max_size = $max_size;
+
+    }
+
+    /**
+     * Check if system could make thumbnail for current file.
+     * @param (string) path to file
+     * @return (boolean)
+     */
+    public function file_support_thumbnail($file){
```

Comment:
What with encrypted files? Those with good image extension but they will fail just in this checking..